python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -283,17 +349,25 @@ def get_bin_pack_residual(
 
     # A most naive bin packing algorithm.
     nodes = copy.deepcopy(node_resources)
+    used = []
     for demand in resource_demands:
         found = False
-        for node in nodes:
+        node = None
```

Comment:
Why do we need this variable?

Reply:
It's needed because python scoping is dumb. If the inner loop doesn't run the second time, we would reuse an old value of `node`. See line 365